src/h5web/visualizations/shared/models.ts

```diff
@@ -23,35 +23,20 @@ export type Size = { width: number; height: number };
 
 export type Domain = [number, number];
 
-export interface IndexAxisConfig {
-  indexDomain: Domain;
-  showGrid?: boolean;
-  scaleType?: never; // invalid
-}
```

Comment:
One could argue that it is now possible to have integer ticks with a scale that is not linear (and therefore not representative of an IndexAxis).

I think it is fine ! It adds more flexiblity and if the caller wants an index axis, it is its responsibility to set it properly.

Reply:
Actually, this conflits with the heatmap: the heatmap cannot accept X/Y axis that are not linear. So I will review this.